### PR TITLE
Code refactoring and tests

### PR DESCRIPTION
**Pull request by machinekoder**
This PR adds the following changes:
* fixes some typos in the `jog_arm_msgs`
* adds another teleop example `spacemouse_to_twist`
* refactoring the jog_arm_server code into separate files for better readability
* fixing clang-format warnings
* improves the performance for joint state updates
* adds `.gitignore` and a `.editorconfig` file
* sets the jog frame header to the default value if nothing else is present
* fixes the `zero_traj_flag` not behaving correctly
* adds a very minimalist 6-DOF test configuration and a sample test case
* adds a timeout for the move group initialization
* fixes jog not working when the joint state is zero (https://github.com/UTNuclearRoboticsPublic/jog_arm/issues/83)

### CODE REVIEW

**Comment by gavanderhoorn** [comment]
@machinekoder: does this overlap with ros-planning/moveit#1244?

**Comment by AndyZe** [comment]
@gavanderhoorn I was planning to go through it manually and add the best/compatible features to the MoveIt! PR. (if that's OK with @machinekoder)

If I do that, I would like to figure out a way to attribute the changes to @machinekoder

**Comment by AndyZe** [comment]
@machinekoder Thanks for being patient. I'm going to start reviewing this now and porting the better changes over to the MoveIt! [pull request](https://github.com/ros-planning/moveit/pull/1244) (if that's OK with you).

File: jog_arm/src/jog_arm/jog_calcs.cpp [hunk 1]
```diff
@@ -614,20 +614,23 @@ namespace jog_arm {
...
-            // Otherwise, commands are in m/s and rad/s
-          else if (parameters_.command_in_type == "speed_units")
-            result[c] = command.deltas[m] * parameters_.publish_period;
-          else
-            ROS_ERROR_STREAM_NAMED(NODE_NAME, "Unexpected command_in_type");
-          goto NEXT_JOINT;
-        }
+      std::size_t c;
+      try {
+        c = jt_state_name_map_.at(incoming_jts_.name[m]);
```
Comment (AndyZe) [inline jog_arm/src/jog_arm/jog_calcs.cpp:619]: I was excited to see this efficiency improvement, but ... there must be an internal loop through jt_state_name_map_ here, so there isn't really an improvement from O(n^2) to O(n), right?

File: jog_arm/src/jog_arm/jog_calcs.cpp [hunk 2]
```diff
@@ -554,28 +557,25 @@ namespace jog_arm {
...
+      catch(const std::out_of_range& e) {
+        ROS_ERROR_STREAM_NAMED(NODE_NAME, "Command joint name unknown.");
+        return false;
+      };
+
+      jt_state_.position[c] = incoming_jts_.position[m];
     }
 
-    return !all_zeros;
+    return true;
```
Comment (AndyZe) [inline jog_arm/src/jog_arm/jog_calcs.cpp]: Unfortunately this breaks the check for all-zero incoming commands -- which means the jogger publishes 0 velocities all the time and I can't execute trajectories anymore.

File: jog_arm/src/jog_arm/jog_calcs.cpp (hunk 1 above)
Comment (gavanderhoorn) [reply jog_arm/src/jog_arm/jog_calcs.cpp:619]: `jt_state_name_map_` is a `std::map` (ie: a ~~hash map~~ balanced tree). Those have ~~O(n)~~ O(log N) cost for lookups. There is no loop.

Comment (AndyZe) [reply jog_arm/src/jog_arm/jog_calcs.cpp:619]: OK, cool. But it's inside a loop:

> for (std::size_t m = 0; m < command.joint_names.size(); ++m)

Anyway, I'll keep this change.

Comment (gavanderhoorn) [reply jog_arm/src/jog_arm/jog_calcs.cpp:619]: `N * O(log N) <= O(N^2)`.

Comment (AndyZe) [reply jog_arm/src/jog_arm/jog_calcs.cpp:619]: Thanks!

**Comment by AndyZe** [comment]
Thank you very much! Lots of good, small contributions, and some big ones too. I love the unit tests.

All of this has been ported to the MoveIt! PR now except the tests. They usually use C++ gtest. I'll wait to see if they want that or not.

**Comment by machinekoder** [comment]
@AndyZe Thanks. Regarding tests: The config/URDF might still be useful even with gtest.

**Comment by machinekoder** [comment]
@AndyZe Any updates on merging the tests?

**Comment by AndyZe** [comment]
They are a part of MoveIt now  :+1: 

https://github.com/ros-planning/moveit/tree/master/moveit_experimental/jog_arm/test

**Comment by AndyZe** [comment]
I guess I will close this PR

**Pull request closed without merging**